index.js

```diff
@@ -156,4 +148,28 @@ function createExtensionMap(exts) {
   return exts;
 }
 
+function isEmpty(object) {
+  return !Object.keys(object).length;
+}
+
+function isString(value) {
+  if (typeof value === 'string') {
+    return true;
+  }
+
+  if (Object.prototype.toString.call(value) === '[object String]') {
+    return true;
+  }
+
+  return false;
+}
+
+function defaultsNotNullish(mainObj, defaultObj) {
+  return filterValues(defaults(mainObj, defaultObj), isNotNullish);
```

Comment:
@phated @jonschlinkert Oh, I mistook. I misunderstood the function of `assignNullish`.
Thank you all for your good helps!